### ISSUE TITLE
One more issue with strings being parsed internally.

### ISSUE THREAD

**Issue by wyattbiker** [didier-v/GDBeautifier#9]
Not sure if this can be easily fixed. Basically it seems to parse strings inside quotes. I think it's when there are quotes inside quotes. E.g. double quotes inside single quotes.

this code
```gdscript
	var ws_patt=r'(?<!\S)--?(?<option>[^\s-][^\s=]*)(?:\s*=\s*(?<value>"[^"]*"|\S+))?|[^\s"]+|"([^"]*)"'
	var x=r'(".+?")'
```

turns into this:
```gdscript
	var ws_patt = r'(?<!\S)--?(?<option>[^\s-][^\s=]*)(?:\s*=\s*(?<value>"[ ^ "]*" | \S + ))? | [ ^ \s"]+|"([ ^ "]*)"'
	var x = r'(". + ?")'
```

**Comment by didier-v** [comment]
Yes, this can become difficult. But, in the specific case of regexes, I may be able to detect the r'...' pattern and ignore what's inside.

**Comment by wyattbiker** [comment]
Actually that would be cool.  r"" or r''  There is also triple """ or ''' that you can ignore what's inside.

This would work for any strings not just regex.

**Comment by didier-v** [comment]
This issue should be fixed with 4a7a5fa10
